ldaptor/test/test_pureldap.py

```diff
@@ -617,12 +617,12 @@ def testToLDAP(self):
             result = klass(*args, **kwargs)
             result = str(result)
             result = map(ord, result)
-            if result!=encoded:
-                raise AssertionError(
-                      "Class %s(*%s, **%s) doesn't encode properly: "
-                      "%s != %s" % (klass.__name__,
-                                    repr(args), repr(kwargs),
-                                    repr(result), repr(encoded)))
+
+            message = (
+                "Class %s(*%r, **%r) doesn't encode properly: "
+                "%r != %r" % (
+                    klass.__name__, args, kwargs, result, encoded))
+            self.assertEqual(encoded, result, message)
```

Comment:
should we eventually migrate to the string.format() way instead of the printf style?

Reply:
As a drive by, we can use string.format()

Careful with string format as you should not pass unsanitized input string as the template.
See http://lucumr.pocoo.org/2016/12/29/careful-with-str-format/

So for simple things, I still use % out of reflex

Reply:
Sanitizing user input should happen before it is used in code anyway.

I had a read through the article, but it isn't anything I wasn't aware of. So long as we are in charge of the code, the objects that are being created by user input, which we are, and we're not doing anything weird with the object or pod when passing it into the string.format() then I'm not worried.

string.format (and other print functions) get a bad rap because they are the things that expose the problem to the end-user or developer anyway. ;)